### ISSUE TITLE
Error searching for projects

### ISSUE THREAD

**Issue by fredrikleroy** [Puzzlepart/prosjektportalen#517]
#### Category

[ ] Enhancement

[x] Bug

[ ] Question

#### Environment

[x] SharePoint Online

[ ] SharePoint 2013

[ ] SharePoint 2016

#### Browser

[x] Chrome

[ ] Internet Explorer

[x] Edge

[ ] Safari

[ ] Other

#### Summary

Project search on front page fails. After the first character is typed in, the input field "freezes" and stops accepting any more characters. Console says error "http://facebook.github.io/react/docs/error-decoder.html?invariant=185". "Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops."

Same thing happens when I click on the menu icons on each of the project cards on the frontpage.

I've changed the GtProjectOwner field to accept multiple entries, but I can't see that this should cause the error.

#### Version

2.1.60 and 2.1.68

#### Expected / Desired Behavior / Question

Normal search functionallity

#### Steps to Reproduce

Start typing in the search box on the front page

![error](https://user-images.githubusercontent.com/17312633/38327409-72863990-3848-11e8-9319-d34adc73f4ca.png)

**Comment by olemp** [comment]
shrinkTitle is from `office-ui-fabric-react`.

```javascript
  DocumentCardTitle.prototype.componentDidUpdate = function () {
        // If we're truncating, make sure the title fits
        if (this.props.shouldTruncate) {
            this._shrinkTitle();
        }
    };
   
    DocumentCardTitle.prototype._shrinkTitle = function () {
        if (this._doesTitleOverflow()) {
            var _a = this.state, truncatedTitleFirstPiece = _a.truncatedTitleFirstPiece, truncatedTitleSecondPiece = _a.truncatedTitleSecondPiece;
            this._isTruncated = true;
            if (!truncatedTitleFirstPiece && !truncatedTitleSecondPiece) {
                this._startTruncation(this.props);
            }
            this.setState({
                truncatedTitleFirstPiece: truncatedTitleFirstPiece.slice(0, truncatedTitleFirstPiece.length - 1),
                truncatedTitleSecondPiece: truncatedTitleSecondPiece.slice(1)
            });
        }
    };

    DocumentCardTitle.prototype._doesTitleOverflow = function () {
        var titleElement = this._titleElement.value;
        if (!titleElement) {
            return false;
        }
        return titleElement.scrollHeight > titleElement.clientHeight + TRUNCATION_VERTICAL_OVERFLOW_THRESHOLD || titleElement.scrollWidth > titleElement.clientWidth;
    };
```

It seems like it's executed in `componentDidUpdate`, and it's updating the state of the component, which might cause an loop. I'll look into possible solutions, @fredrikleroy.

**Comment by olemp** [comment]
What's the full title of the project in the screenshot above, @fredrikleroy?

**Comment by olemp** [comment]
Attempt to fix https://github.com/Puzzlepart/prosjektportalen/commit/06ec161fb03668849e736bf5ec0886d284a5592d

**Comment by fredrikleroy** [comment]
Full title **"Digitalisering, sluttvurdering, dokumentasjon og IKT"**

**Comment by fredrikleroy** [comment]
Thanks for the quick response :)

**Comment by fredrikleroy** [comment]
The portal currently holds around 70 projects, so you're probably right that this is related to many shrink operations.... I'll let u know how it works if I get the time to build it before next release. Thanks again.

**Comment by tarjeieo** [comment]
Looks to be fixed in the latest version (2.1.75) - feel free to verify @fredrikleroy ! :bowtie:
